apps/inviwodome/src/inviwodome.cpp

```diff
@@ -0,0 +1,370 @@
+
+    std::mutex commandsMutex;
+    std::vector<inviwo::SgctCommand> commands;
+
+    void setCamerasToSGCT(inviwo::ProcessorNetwork& net) {
+        inviwo::LambdaNetworkVisitor visitor{[](inviwo::Property& p) {
+            if (auto camera = dynamic_cast<inviwo::CameraProperty*>(&p)) {
+                camera->setCamera("SGCTCamera");
+            }
+        }};
+        net.accept(visitor);
+    }
+
+    void initOpenGL(GLFWwindow* shared) {
+        // Tell GLFW that we already have a shared context;
+        inviwo::CanvasGLFW::provideExternalContext(shared);
+        inviwo::SGCTModule::startServer = false;
+
+        inviwo::initializePythonModules();
+
+        // Initialize all modules non Qt modules
+        auto filter = [](const inviwo::ModuleContainer& m) {
+            return m.identifier().ends_with("qt") || m.identifier().starts_with("qt");
+        };
+        inviwo::util::registerModulesFiltered(app.getModuleManager(), filter,
+                                              app.getSystemSettings().moduleSearchPaths_.get());
+
+        // Uncomment to debug frame timing issues.
+        // app.getModuleByType<inviwo::GLFWModule>()->setWaitForOpenGL(false);
+
+        state.createShader();
+
+        if (sgct::Engine::instance().isMaster()) {
+            for (auto& win : sgct::Engine::instance().windows()) {
+                state.setupInteraction(win->windowHandle());
+            }
+        }
+        GLint maxDrawBuffers = 8;
+        glGetIntegerv(GL_MAX_DRAW_BUFFERS, &maxDrawBuffers);
+        drawBuffers.resize(static_cast<size_t>(maxDrawBuffers), GL_NONE);
+
+        if (sgct::Engine::instance().isMaster()) {
+            app.getWorkspaceManager()->load(
+                app.getPath(inviwo::PathType::Workspaces, "/boron.inv"));
+            setCamerasToSGCT(*app.getProcessorNetwork());
+        } else {
+            // Only the master node needs to have any widgets.
+            auto* wf = app.getProcessorWidgetFactory();
+            auto wfKeys = wf->getKeys();
+            for (auto& key : wfKeys) {
+                wf->unRegisterObject(wf->getFactoryObject(key));
+            }
+        }
+    };
+
+    void preSync() {
+        TRACY_ZONE_SCOPED_NC("Process Front", 0xAA0000);
+
+        // We check the front queue before every frame. No need for setPostEnqueueFront
+        app.processFront();
+    };
+
+    auto encode() -> std::vector<std::byte> { return syncServer->getEncodedCommandsAndClear(); }
+
+    void decode(const std::vector<std::byte>& bytes) {
+        auto tmp = inviwo::util::decode(bytes);
+        std::scoped_lock lock{commandsMutex};
+        commands.insert(commands.end(), tmp.begin(), tmp.end());
+    }
+
+    void postSyncPreDraw() {
+        std::scoped_lock lock{commandsMutex};
+        syncClient->applyCommands(commands);
+        commands.clear();
+    };
+
+    void draw(const sgct::RenderData& renderData) {
+        TRACY_ZONE_SCOPED_NC("Draw", 0xAAAA00);
```

Comment:
are `TRACY` macros defined even if Tracy is not enabled?

Reply:
Yes